sdk/core/Azure.Core/src/Response.cs

```diff
@@ -129,9 +110,7 @@ public virtual BinaryData Content
         /// <param name="response">The HTTP response.</param>
         /// <returns>A new instance of <see cref="Response{T}"/> with the provided value and HTTP response.</returns>
         public static Response<T> FromValue<T>(T value, Response response)
-        {
-            return new ValueResponse<T>(response, value);
-        }
+            => new AzureCoreResponse<T>(value, response);
 
         /// <summary>
         /// Returns the string representation of this <see cref="Response"/>.
```

Comment:
TODO: Remove DisposeStreamIfNotBuffered when we update the transport response implementations.